### ISSUE TITLE
Docker Registry API

### ISSUE THREAD

**Issue by nVitius** [api-stack/api-stack#40]
The V1 API was deprecated not too long ago. It would be great to have a swagger definition for the V2 API. It looks like most (all?) are contributed by @kinlane. Do you have any tips on how to get started writing one? I would like to help. Also, are there any guidelines as to how they should be written?

**Comment by kinlane** [comment]
Yeah...I would like to have the latest too. I have a documentation script that generated the first draft of v1.16. I checked and it doesn't run on the new version. they changed the HTML. I can probably have updated at some point to today, and get a first draft of the surface area for 1.21 - https://docs.docker.com/engine/reference/api/docker_remote_api_v1.21/ - Will still need hand polishing, but would be good start.

**Comment by nVitius** [comment]
Ohhh. You parse the HTML to generate the documentation. That's a really good idea; I like it. Do you have those somewhere?

**Comment by kinlane** [comment]
Yeah most APIs are still static HTML, and about 70% of the time it is consistent enough HTML to parse, and auto generate a machine readable spec. I only get the essentials usually, but sometimes it saves some serious data entry on parameter descriptions, etc. I don't have my scripts publicly...I just locally develop, and run as I have the time. I will prioritize docker...so I can get new version up. Important. If there are other APIs you'd like to see, let me know.
